src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs

```diff
@@ -1160,14 +1160,22 @@ private void BeforeQueryStatusForAddPackageForSolutionDialog(object sender, Even
                 {
                     isConsoleBusy = ConsoleStatus.Value.IsBusy;
                 }
-                // Enable the 'Manage NuGet Packages For Solution' dialog menu
-                // - if the console is NOT busy executing a command, AND
-                // - if the solution exists and not debugging and not building AND
-                // - if there are NuGetProjects. This means there are loaded, supported projects.
-                command.Enabled =
-                    IsSolutionExistsAndNotDebuggingAndNotBuilding() &&
-                    !isConsoleBusy &&
-                    await SolutionManager.Value.DoesNuGetSupportsAnyProjectAsync();
+
+                try
+                {
+                    // Enable the 'Manage NuGet Packages For Solution' dialog menu
+                    // - if the console is NOT busy executing a command, AND
+                    // - if the solution exists and not debugging and not building AND
+                    // - if there are NuGetProjects. This means there are loaded, supported projects.
+                    command.Enabled =
+                        IsSolutionExistsAndNotDebuggingAndNotBuilding() &&
+                        !isConsoleBusy &&
+                        await SolutionManager.Value.DoesNuGetSupportsAnyProjectAsync();
+                }
+                catch (Exception e)
```

Comment:
The `BeforeQueryStatusForAddPackageForSolutionDialog` handler is something that's called from another component, do they have the telemetry?

Reply:
It's called from [line#315](https://github.com/NuGet/NuGet.Client/blob/5ce9c44823a8930dd0bf57adbf5c32914d8e4ae3/src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs#L315) and that method `AddMenuCommandHandlersAsync` itself is called from line 159.

https://github.com/NuGet/NuGet.Client/blob/5ce9c44823a8930dd0bf57adbf5c32914d8e4ae3/src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs#L151-L160

I don't see any specific telemetries for calling method, I can see below events right before exception happen, not sure if it helps:

![image](https://user-images.githubusercontent.com/8766776/107692138-3c0c5200-6c61-11eb-9d17-9d06229419b0.png)

![image](https://user-images.githubusercontent.com/8766776/107693098-86da9980-6c62-11eb-9f72-2e2dd54a6cda.png)

Reply:
Thanks for looking at that. 

What are you trying to get through the telemetry? It has the same concern as the earlier implementation. We should try to fix it rather than workaround it.

Reply:
> Thanks for looking at that.
> 
> What are you trying to get through the telemetry? It has the same concern as the earlier implementation. We should try to fix it rather than workaround it.

I thought [we're interested](https://github.com/NuGet/NuGet.Client/pull/3891#issuecomment-776967793) in how it affect customer before fix it. 
Ok. In that case I'll rather fix it than telemetry and workaround.

Reply:
I think @zivkan's point was that we should try to estimate the impact because you had claimed that it doesn't affect customers, so he was just asking whether we have more than just a guess there.

Reply:
I think the issue looks like a real product bug that can affect customers, hence I disagree with claims such as "Customers wouldn't experience any change".